receiver/stanzareceiver/receiver.go

```diff
@@ -34,12 +44,51 @@ var _ component.LogsReceiver = (*stanzareceiver)(nil)
 
 // Start tells the receiver to start
 func (r *stanzareceiver) Start(ctx context.Context, host component.Host) error {
-	// TODO
-	return nil
+	r.Lock()
+	defer r.Unlock()
+	err := componenterror.ErrAlreadyStarted
+	r.startOnce.Do(func() {
+		err = nil
+		rctx, cancel := context.WithCancel(ctx)
+		r.cancel = cancel
+		r.logger.Info("Starting observiq receiver")
+
+		if obsErr := r.agent.Start(); obsErr != nil {
+			err = fmt.Errorf("start observiq: %s", err)
+		}
+
+		r.wg.Add(1)
+		go func() {
+			defer r.wg.Done()
+			for {
+				select {
+				case <-rctx.Done():
+					return
+				case obsLog, ok := <-r.emitter.logChan:
+					if !ok {
+						continue
+					}
+					if consumeErr := r.consumer.ConsumeLogs(ctx, convert(obsLog)); consumeErr != nil {
```

Comment:
This conversion routine is pretty inefficient in that there is an entire `pdata.Logs` created for each log record from Stanza.

Is it reasonable to expect multiple log records in close proximity with the same resource attributes?  If so you could pull off the channel and batch until the channel is empty and convert many at once into a single pdata.Logs (with potentially multiple pdata.ResourceLogs).  it isn't very easy to compare Resource since they are maps, and maybe doing that would overcome any performance advantages of batching.  But still something to consider.

Reply:
> Is it reasonable to expect multiple log records in close proximity with the same resource attributes?

Yes, quite often. I think you have a good point, and suspect that this is improvable.

I did actually take a stab at this earlier on, but did not see an improvement in the benchmarks at that time. I think it'll be worth taking a closer look though.

For now, I view the current benchmark results as acceptable, so it's up to you whether this is a necessary improvement at this point. My preference, for what it's worth, would be to leave batching out of the initial implementation and consider it a possible future optimization.

Reply:
I don't consider it essential for an initial version, and it is definitely not a trivial thing to do.